tsl/test/expected/detach_chunk.out

```diff
@@ -0,0 +1,179 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+\c :TEST_DBNAME :ROLE_SUPERUSER
+GRANT CREATE ON DATABASE :"TEST_DBNAME" TO :ROLE_DEFAULT_PERM_USER;
+SET ROLE :ROLE_DEFAULT_PERM_USER;
+-- Create test tables and hypertables
+CREATE TABLE devices(id int PRIMARY KEY);
+INSERT INTO devices VALUES (1), (2), (3);
+CREATE TABLE detach_test(id int, time timestamptz not null, device int, temp float);
+CREATE INDEX detach_test_device_idx ON detach_test (device);
+ALTER TABLE detach_test
+    ADD CONSTRAINT detach_test_temp_check CHECK (temp > 0),
+    ADD CONSTRAINT detach_test_device_fkey FOREIGN KEY (device) REFERENCES devices(id),
+    ADD CONSTRAINT detach_test_id_time_unique UNIQUE (id, time);
+SELECT * FROM create_hypertable('detach_test', 'time', 'id', 2);
+ hypertable_id | schema_name | table_name  | created 
+---------------+-------------+-------------+---------
+             1 | public      | detach_test | t
+(1 row)
+
+CREATE TABLE detach_test_ref (
+    id int PRIMARY KEY,
+    ref_id int,
+    ref_time timestamptz,
+    FOREIGN KEY (ref_id, ref_time) REFERENCES detach_test(id, time)
+);
+INSERT INTO detach_test VALUES
+    (1, '2025-06-01 05:00:00-8', 1, 23.4),
+    (2, '2025-06-15 05:00:00-8', 2, 24.5),
+    (3, '2025-06-30 05:00:00-8', 3, 25.6);
+-- Get chunk information for testing
+SELECT
+    chunk_id AS "CHUNK_ID",
+    hypertable_id AS "HYPERTABLE_ID",
+    schema_name AS "CHUNK_SCHEMA",
+    table_name AS "CHUNK_TABLE",
+    schema_name || '.' || table_name AS "CHUNK_NAME"
+FROM _timescaledb_functions.show_chunk((SELECT show_chunks('detach_test') LIMIT 1)); \gset
+ CHUNK_ID | HYPERTABLE_ID |     CHUNK_SCHEMA      |   CHUNK_TABLE    |               CHUNK_NAME               
+----------+---------------+-----------------------+------------------+----------------------------------------
+        1 |             1 | _timescaledb_internal | _hyper_1_1_chunk | _timescaledb_internal._hyper_1_1_chunk
+(1 row)
+
+-- Test successful detach
+CALL detach_chunk(:'CHUNK_NAME');
+-- Verify chunk is detached:
+SELECT count(*) FROM _timescaledb_catalog.chunk WHERE hypertable_id = :'HYPERTABLE_ID';
+ count 
+-------
+     2
```

Comment:
It is not clear what number of chunks were there before, so might be good to add a check of the number of chunks before as well.